Assets/Gothic-UnZENity-Core/Scripts/GameManager.cs

```diff
@@ -0,0 +1,185 @@
+        public VobSoundCullingManager SoundCulling => _soundCullingManager;
+
+		// ReSharper disable Unity.PerformanceAnalysis
+		private void Load()
+		{
+			// If the Gothic installation directory is not set, show an error message and exit.
+			if (!_settings.CheckIfGothic1InstallationExists())
+			{
+				invalidInstallationPathMessage.SetActive(true);
+				return;
+			}
+
+			// Otherwise, continue loading Gothic.
+            ResourceLoader.Init(_settings.GothicIPath);
+	
+            _gameMusicManager.Init();
+            
+			GUZBootstrapper.BootGothicUnZENity(config, _settings.GothicIPath, _settings.GothicILanguage);
+			_gameSceneManager.LoadStartupScenes();
+
+			if (config.enableBarrierVisual)
+			{
+				GlobalEventDispatcher.WorldSceneLoaded.AddListener(() =>
+				{
+					_barrierManager.CreateBarrier();
+				});
+			}
+		}
+
+		private void Awake()
+		{
+			GameGlobals.Instance = this;
+			LookupCache.Init();
+
+			_textureManager = GetComponent<TextureManager>();
+			_fontManager = GetComponent<FontManager>();
+			_settings = GameSettings.Load();
+			_fileLoggingHandler = new FileLoggingHandler(_settings);
+			_gameLoadingManager = new LoadingManager();
+			_meshCullingManager = new VobMeshCullingManager(config, this);
+			_soundCullingManager = new VobSoundCullingManager(config);
+			_barrierManager = new BarrierManager(config);
+			_stationaryLightsManager = new StationaryLightsManager();
+			_xrSimulatorManager = new XRDeviceSimulatorManager(config);
+			_gameTimeManager = new GameTime(config, this);
+			_skyVisualManager = new SkyManager(config, _gameTimeManager, _settings);
+			_gameMusicManager = new MusicManager(config);
+			_gameSceneManager = new GUZSceneManager(config, _gameLoadingManager, xrInteractionManager);
+			_npcRoutineManager = new RoutineManager(config);
+		}
+
+		private void Start()
+		{
+			ZenKit.Logger.Set(config.zenkitLogLevel, Logging.OnZenKitLogMessage);
+			DirectMusic.Logger.Set(config.directMusicLogLevel, Logging.OnDirectMusicLogMessage);
+
+			_fileLoggingHandler.Init();
+			_gameLoadingManager.Init();
+			_meshCullingManager.Init();
+			_soundCullingManager.Init();
+			_gameTimeManager.Init();
+			_skyVisualManager.Init();
+			_xrSimulatorManager.Init();
+			_gameSceneManager.Init();
+			_npcRoutineManager.Init();
+
+			// Just in case we forgot to disable it in scene view. ;-)
+			invalidInstallationPathMessage.SetActive(false);
+			
+			// Load the player controller upon MainMenu loaded
+            GlobalEventDispatcher.MainMenuSceneLoaded.AddListener(delegate
+            {
+                GUZContext.InteractionAdapter.CreatePlayerController(SceneManager.GetActiveScene());
+            });
+		}
+
+		private void Update()
+		{
+			_gameSceneManager.Update();
+
+			if (_isInitialised)
+			{
+				return;
+			}
+
+			_isInitialised = true;
+			Load();
+		}
+
+		private void FixedUpdate()
+		{
+			_barrierManager.FixedUpdate();
```

Comment:
Barrier manager should take care of it. Do you really want to centralize these Unity calls? I know your point about having nearly 0 managers as a Component, but I don't think this will scale properly. Let's discuss.